src/querier_set.rs

```diff
@@ -0,0 +1,75 @@
+use priv_prelude::*;
+use future_utils::mpsc::{self, UnboundedSender, UnboundedReceiver};
+use std::collections::hash_map;
+use std::hash::Hasher;
+
+#[derive(Default, Debug)]
+pub struct TcpQuerierSet {
+    queriers: HashMap<u64, Arc<TcpQuerier>>,
+    txs: Vec<UnboundedSender<Arc<TcpQuerier>>>,
+}
+
+impl TcpQuerierSet {
+    pub fn add_querier(&mut self, querier: impl TcpQuerier + Hash) {
+        let querier = Arc::new(querier);
+        self.txs.retain(|tx| {
+            tx.unbounded_send(querier.clone()).is_ok()
+        });
+        let mut hasher = hash_map::DefaultHasher::new();
+        querier.hash(&mut hasher);
+        let hash = hasher.finish();
+        let _ = self.queriers.insert(hash, querier);
+    }
+
+    pub fn remove_querier<T: Hash>(&mut self, querier: &T) {
+        let mut hasher = hash_map::DefaultHasher::new();
+        querier.hash(&mut hasher);
+        let hash = hasher.finish();
+        let _ = self.queriers.remove(&hash);
+    }
+
+    pub fn queriers(&mut self) -> UnboundedReceiver<Arc<TcpQuerier>> {
```

Comment:
Rendezvous connects are a long process that could take up to two minutes. Any additions to the set of queriers should be visible to the rendezvous connect code while it's in-progress. This could be particularly important to programs that are booting up in parallel, adding their queriers while also performing their first connections.